### PR TITLE
Improve `function-name-case` document

### PR DESCRIPTION
**Pull request by ybiquitous**
- Use dashed idents in examples
- Remove incorrect example

> Which issue, if any, is this issue related to?

Part of #5523

> Is there anything in the PR that needs further explanation?

- Use dashed idents in examples
- Remove incorrect example

### CODE REVIEW

File: lib/rules/function-name-case/README.md [hunk 1]
```diff
@@ -115,7 +115,7 @@ a {
 
 ## Optional secondary options
 
-### `ignoreFunctions: ["/regex-as-string/", /regex/, "non-regex"]`
+### `ignoreFunctions: ["/regex/", /regex/, "non-regex"]`
```
Comment (ybiquitous) [inline lib/rules/function-name-case/README.md:118]: [note] `regex-as-string` rarely used.

File: lib/rules/function-name-case/README.md [hunk 2]
```diff
@@ -124,36 +124,29 @@ For example, with `"lower"`.
...
 a {
-  color: sOmE-FuNcTiOn();
+  color: --sOmE-FuNcTiOn();
 }
 ```
 
 <!-- prettier-ignore -->
 ```css
 a {
-  color: some-other-function();
```
Comment (ybiquitous) [inline lib/rules/function-name-case/README.md:142]: [note] `some-other-function` is not a problem.

File: lib/rules/function-name-case/README.md [hunk 3]
```diff
@@ -124,36 +124,29 @@ For example, with `"lower"`.
 Given:
 
 ```json
-["some-function", "/^get.*$/"]
+["--some-function", "/^--get.*$/"]
 ```
 
 The following patterns are considered problems:
```
Comment (ybiquitous) [inline lib/rules/function-name-case/README.md:130]: [ask] Do we need to give examples of considered problems in `ignoreFunctions`?

Comment (jeddy3) [reply lib/rules/function-name-case/README.md:130]: Narh, we don't need to. We can just show the rule being made more permissive. For example, by showing a pattern that was a problem previously without the option being used:

e.g. before the secondary options header we can show `--FOO()` being a problem, and then:

Given:

```
["--FOO"]
```

It no longer being an problem. 

I think we can just take the problem section out in this pull request, and save the suggestions above (showing the permissiveness and using `foo`) for another issue/pr.

Comment (ybiquitous) [reply lib/rules/function-name-case/README.md:130]: Thanks for the explanation. I understand and agree. 👍🏼
Then, I'll keep the PR change as-is.

**Review by jeddy3 — APPROVED** [review]

**Pull request merged**